src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1844,60 +1869,17 @@ protected TypeSymbol DecodeFieldSignature(ref BlobReader signatureReader, out bo
 
             try
             {
-                // See if there is a Volatile modifier.
+                var isVolatileFound = false;
                 SignatureTypeCode typeCode;
-                ArrayBuilder<ModifierInfo<TypeSymbol>> customModifierBuilder = null;
```

Comment:
>ArrayBuilder<ModifierInfo<TypeSymbol>> customModifierBuilder = null; [](start = 16, length = 68)

I would revert changes in this function, it doesn't look like we want any behavior change here.